stingray/pulse/pulsar.py

```diff
@@ -263,18 +263,24 @@ def fold_events(times, *frequency_derivatives, **opts):
     profile_err : array of floats
         The uncertainties on the pulse profile
     """
-    mode = _default_value_if_no_key(opts, "mode", "ef")
-    nbin = _default_value_if_no_key(opts, "nbin", 16)
-    weights = _default_value_if_no_key(opts, "weights", 1)
+
+    mode = dict.pop(opts, "mode", "ef")
+    nbin = dict.pop(opts, "nbin", 16)
+    weights = dict.pop(opts, "weights", 1)
     # If no key is passed, *or gti is None*, defaults to the
     # initial and final event
-    gti = _default_value_if_no_key(opts, "gti", None)
+    gti = dict.pop(opts, "gti", None)
```

Comment:
Actually, sorry, I think you can just use `opts.pop("gti", None)`

Reply:
I've made the changes